multiplexer/go.mod

```diff
@@ -0,0 +1,201 @@
+module github.com/celestiaorg/celestia-app/multiplexer
```

Comment:
idk if `module github.com/celestiaorg/celestia-app/mux` is more "go idiomatic", but it doesn't really matter

Reply:
[uber nit] I prefer `multiplexer` as the package name over `mux`